### PR TITLE
Read schema from Python package (#169 )

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
if python is set & taipy.core package is installed and a schema is found then use it
v 0.4.4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -330,6 +330,12 @@
           "default": "main.py",
           "description": "%taipy.config.settings.config.mainPythonFile.desc%"
         }
+        ,
+        "taipyStudio.config.useShemaFromPackage": {
```
Comment (dinhlongviolin1) [inline package.json]: First time seeing a Shema

File: package.json [hunk 2]
```diff
@@ -330,6 +330,12 @@
...
+          "type": "boolean",
+          "default": true,
+          "description": "%taipy.config.settings.config.useShemaFromPackage.desc%"
```
Comment (dinhlongviolin1) [inline package.json]: Shema

File: package.nls.json [hunk 3]
```diff
@@ -40,5 +40,6 @@
     "taipy.config.settings.diagram.scenario.color.desc": "The color for the Scenario Diagram element.",
     "taipy.config.settings.diagram.scenario.icon.desc": "The icon for the Scenario Diagram element.",
     "taipy.config.settings.status.timeout.desc": "The timeout for Status Bar Item (ms).",
-    "taipy.config.settings.config.mainPythonFile.desc": "The name of the main Python file (module: __main__)."
+    "taipy.config.settings.config.mainPythonFile.desc": "The name of the main Python file (module: __main__).",
+    "taipy.config.settings.config.useShemaFromPackage.desc": "Use configuration schemas from the taipi.* packages."
```
Comment (dinhlongviolin1) [inline package.nls.json]: taipi or taipy

File: src/schema/validation.ts [hunk 4]
```diff
@@ -13,11 +13,29 @@
...
+import { workspace } from "vscode";
+
+import { getFilesFromPythonPackages } from "../utils/utils";
+import { TAIPY_STUDIO_SETTINGS_NAME } from "../utils/constants";
 
 let validationSchema: Schema;
 export const getValidationSchema = async () => {
   if (!validationSchema) {
-    validationSchema = await import("../../schemas/config.schema.json");
+    if (workspace.getConfiguration(TAIPY_STUDIO_SETTINGS_NAME).get("config.useShemaFromPackage", true)) {
```
Comment (dinhlongviolin1) [inline src/schema/validation.ts]: shema here too

File: src/utils/utils.ts [hunk 5]
```diff
@@ -35,3 +37,17 @@ export const getDefaultConfig = (webview: Webview, extensionUri: Uri) => {
 };
 
 export const getPositionFragment = (pos: Position) => `L${pos.line + 1}C${pos.character}`;
+
+export const getFilesFromPythonPackages = (file: string, packages: string[]) => {
+  const config = workspace.getConfiguration("python");
+  const pythonPath = config.get("pythonPath", "python");
+  return new Promise<Record<string, string>>((resolve, reject) => {
+    exec(`"${pythonPath}" "${join(__dirname, "python", "find_file_in_package.py")}" "${file}" "${packages.join('" "')}"`, (error, stdout, stderr) => {
```
Comment (dinhlongviolin1) [inline src/utils/utils.ts]: We will see whether __dirname could be used after deploying. I guess it should work fine

**Review by dinhlongviolin1 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (FredLL-Avaiga) [reply package.json]: tx

File: package.nls.json (hunk 3 above)
Comment (FredLL-Avaiga) [reply package.nls.json]: good catch

File: src/utils/utils.ts (hunk 5 above)
Comment (FredLL-Avaiga) [reply src/utils/utils.ts]: Yes it will as we are in a node.js process

Comment (FredLL-Avaiga) [reply src/utils/utils.ts]: Copy/paste from your code :-)

File: package.json [hunk 6]
```diff
@@ -218,11 +218,11 @@
       "view/item/context": [
         {
           "command": "taipy.perspective.show",
-          "when": "view == taipy-configs || viewItem != default && view == taipy-config-pipelines || viewItem != default && view == taipy-config-scenarios"
+          "when": "view == taipy-configs || viewItem != default && view =~ /taipy-config-(pipelines|scenarios)/"
         },
         {
           "command": "taipy.diagram.addNode",
-          "when": "viewItem != default && view == taipy-config-datanodes || viewItem != default && view == taipy-config-tasks || viewItem != default && view == taipy-config-pipelines"
+          "when": "taipy.config.diagram.visible && viewItem != default && view =~ /taipy-config-(datanodes|tasks|pipelines)/"
```
Comment (FabienLelaquais) [inline package.json:225]: Well done!

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**